core/src/main/java/org/apache/iceberg/rest/InMemoryPlanningState.java

```diff
@@ -43,10 +44,14 @@ class InMemoryPlanningState {
 
   private final Map<String, List<FileScanTask>> planTaskToFileScanTasks;
   private final Map<String, String> planTaskToNext;
+  private final Set<String> ongoingAsyncPlans;
+  private final Set<String> failedAsyncPlans;
```

Comment:
can we just maintain a single store for storing states like stateMachine for example

Map<String, String> planState;

on submitting it to the executor thread status for plan ID : 'SUBMITTED'

Now based on execution it can go to these terminal states
SUBMITTED -> FAILED
SUBMITTED -> CANCELLED
SUBMITTED -> COMPLETED